runtime/doc/api.txt

```diff
@@ -1949,12 +1949,35 @@ nvim_get_option_info({name})                          *nvim_get_option_info()*
     • commalist: List of comma separated values
     • flaglist: List of single char flags
 
+    The last set information applies to the local value in the current buffer
+    or window if it is available, otherwise the global value information is
+    returned.
+
     Parameters: ~
       • {name}  Option name
 
     Return: ~
         Option Information
 
+nvim_get_option_info_ex({name}, {*opts})           *nvim_get_option_info_ex()*
+    Gets the option information for one option from arbitrary buffer or window
+
+    This function is similar to nvim_get_option_info except that it accepts an
+    additional options table {opts}, allowing to specify option scope, buffer
+    number or window used for retrieving the last set location information.
+
+    Parameters: ~
+      • {name}  Option name
+      • {opts}  Optional parameters
+                • scope: One of "global" or "local". Analogous to |:setglobal|
```

Comment:
Should it also support empty string "" which mimics `:echo &foo` (as opposed to `:echo &g:foo` and `:echo &l:foo` ) ? Or is that already supported by omitting `opts.scope` ?

Reply:
I'm assuming the default value is "global"?

Reply:
I think it's better to assume explicit intent when specifying scope, so
1. if `scope = 'global'`, _only_ return the global value (behave like `:setglobal`)
2. if `scope = 'local'`, _only_ return the local value (empty if not set) (behave like `:setlocal`)
3. if `scope = nil`, try local then fallback to global (behave like `:set`)

Reply:
Done. I think that the tests now cover all the cases mentioned by @clason.